vimeo/vimeo-networking-java
# ADF-39 Remove Deprecated classes/methods/fields

#### Ticket
[ADF-39](https://vimean.atlassian.net/browse/ADF-39)

#### Ticket Summary
We should remove deprecated classes, methods, and fields in preparation for 2.0.0

#### Implementation Summary
This is straightforward, so I am just listing the notables:
* I kept `Embed` as Deprecated, since the API hasn't released the new Embed representation (even though we have it in this library). Once they do, it will require a version bump. I clarified in javadocs about this.
* I bumped the version to 3.3.5 based on the API changelog around the reason we added a privacy deserializer, https://vimean.atlassian.net/wiki/display/API/Changelog and then removed the custom deserializer
* I modified the Configuration.Builder so that you cannot pass in a base URL. You can change the base url only by grabbing the current Configuration first.
* I made VideoFile abstract, and subclassed it into DashVideoFile, HlsVideoFile, and ProgressiveVideoFile
* I updated the Readme to reflect changes

#### How to Test
Test via Android PR forthcoming:
* Verify fetching a video without field filters (edit the video settings) does not crash, since the VideoFileDeserializer was removed
* Test Privacy values are correctly deserialized (remove the filtering of privacy add/download)
* Test changing the base url via the admin panel, use stetho to sniff
